### PR TITLE
runfix: message list correct key

### PR DESCRIPTION
**Pull request by kamil-au**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### CODE REVIEW

File: src/script/components/MessagesList/index.tsx [hunk 1]
```diff
@@ -179,9 +179,10 @@ const MessagesList: React.FC<MessagesListParams> = ({
     const isLastDeliveredMessage = lastDeliveredMessage?.id === message.id;
 
     const visibleCallback = getVisibleCallback(conversation, message);
+    const key = (message.id || 'message-') + message.timestamp();
```
Comment (kamil-au) [inline src/script/components/MessagesList/index.tsx:182]: For some messages `id` is undefined, and tine `timestamp()` is ko.observable so it has to be accessed this way not just `timestamp`.

File: src/script/components/MessagesList/Message/index.tsx [hunk 2]
```diff
@@ -107,12 +107,10 @@ const Message: React.FC<
 
   const content = <MessageWrapper {...props} hasMarker={markerType !== MessageMarkerType.NONE} />;
   const wrappedContent = props.onVisible ? <InViewport onVisible={props.onVisible}>{content}</InViewport> : content;
-
   return (
     <div
       className={`message ${isMarked ? 'message-marked' : ''}`}
       ref={messageElementRef}
-      key={message.id}
```
Comment (kamil-au) [inline src/script/components/MessagesList/Message/index.tsx:115]: This one seems to be not required as we don't have a list here.

**Review by thisisamir98 — APPROVED** [review]

File: src/script/components/MessagesList/index.tsx (hunk 1 above)
Comment (BartDyb) [reply src/script/components/MessagesList/index.tsx:182]: maybe ``message.id || `message-${message.timestamp()}` ``?

**Review by BartDyb — APPROVED** [review]

File: src/script/components/MessagesList/index.tsx (hunk 1 above)
Comment (kamil-au) [reply src/script/components/MessagesList/index.tsx:182]: It will provide duplicated key, as the system messages have the same string as `id`

**Review by atomrc — COMMENTED** [review]
Good catch, thanks for fixing it :pray:

**Pull request merged**